src/App.js

```diff
@@ -2,39 +2,58 @@ import "./App.css";
 import Cell from "./Cell.js";
 import NumberInput from "./NumberInput";
 import { useEffect, useState } from "react";
+import { v4 as uuidv4 } from "uuid";
 
 function App() {
   const [numbers, setNumbers] = useState([[], [], [], [], [], [], [], [], []]);
-  let selectedCell = [0, 0];
+  const [isLoading, setIsLoading] = useState(false);
+  let selectedCell = [];
 
   useEffect(() => {
     let url = "https://sugoku.herokuapp.com/board?difficulty=easy";
-
+    setIsLoading(true);
     fetch(url)
       .then((resp) => resp.json())
       .then((data) => {
         setNumbers(data.board);
+        localStorage.setItem("currentSudoku", JSON.stringify(data.board));
+        setIsLoading(false);
       });
   }, []);
 
   function renderCellRow(rowNumber) {
-    const numbArray = numbers[rowNumber].map((number, index) => {
+    if (isLoading || numbers === null || numbers === undefined) {
       return (
-        <Cell
-          key={index}
-          value={number}
-          id={[rowNumber, index]}
-          onClick={handleCellClick}
-        />
+        <>
+          <p>Loading...</p>
+        </>
       );
-    });
-    return numbArray;
+    } else {
+      let mask = JSON.parse(localStorage.getItem("currentSudoku"));
+      console.log("mask: " + mask);
```

Comment:
I was told that we should remove console logs in the final code. ;-)))

Reply:
Ah shoot, I copy pasted it back in XD